src/main/java/org/spongepowered/api/data/DataRegistration.java

```diff
@@ -40,8 +41,12 @@
      * the desired {@link DataManipulator} type to be registered.
      *
      * @return The new builder instance
+     * @deprecated Use the {@link DataGenerator} instead, this is the new
+     *    method of generating {@link DataManipulator} implementations and
```

Comment:
Why would this be removed from the API? The point of dynamic class generation is not to wholeheartedly replace its method of building (since sponge implementation is likely to build with it).